### PR TITLE
StringPermutation.java-usha

### PR DESCRIPTION
**Pull request by UUsha**
input is null
cat,tAC is true
ball, lab is false

### CODE REVIEW

**Comment by chandramouly** [comment]
Great, amazing solution... O(n) with no extra space.. good job Usha! way to go..

File: class7/isStringPerm.java [hunk 1]
```diff
@@ -1,6 +1,59 @@
...
+
+public class StringPermutation {
+
+	public StringPermutation() {
+		// TODO Auto-generated constructor stub
+	}
+	
+	public static boolean isStringPerm(String string1, String string2){
+		
+		int[] charCountString = new int[30];
```
Comment (manojgo) [inline class7/isStringPerm.java:10]: @chandramouly , @UUsha, 

This is the extra space usage.

**Comment by manojgo** [comment]
@chandramouly , @UUsha 

There is extra space used: int[] charCountString = new int[30];

**Comment by chandramouly** [comment]
But still this space is always going to be constant and can be ignored.

**Pull request closed without merging**